src/scales/scale.linearbase.js

```diff
@@ -23,6 +24,10 @@ function generateTicks(generationOptions, dataRange) {
 	var precision = generationOptions.precision;
 	var spacing, factor, niceMin, niceMax, numSpaces;
 
+	// sanitize dataRangee to MAX_PRECISION
+	dataRange.min = Math.floor(dataRange.min * MAX_PRECISION) / MAX_PRECISION;
+	dataRange.max = Math.ceil(dataRange.max * MAX_PRECISION) / MAX_PRECISION;
```

Comment:
`generateTicks` is called with `this` as `dataRange`, so we may not want to modify this object. Instead, we should use local variables:

```javascript
var rmin = Math.floor(dataRange.min * MAX_PRECISION) / MAX_PRECISION;
var rmax = Math.ceil(dataRange.max * MAX_PRECISION) / MAX_PRECISION;
```

It would actually save a few bits and perform better I guess.